### ISSUE TITLE
FEATURE REQUEST:  kubectl unset command

### ISSUE THREAD

**Issue by xilabao** [kubernetes/kubernetes#39775]
Since we have kubectl set command to configure the resources. Should we add the unset command to cancel some configs?

**Comment by snowsky** [comment]
Good point! Probably we can still use set command but add some parameter, like -u?

**Comment by xilabao** [comment]
@pwittrock @kargakis

**Comment by kargakis** [comment]
@kubernetes/sig-cli-misc

**Comment by rothgar** [comment]
I'm not sure `unset` would make sense with the current options. Currently set can only set an image of a pod template but unless you're using a deployment (with history tracking enabled) there wouldn't be anything you will be able to unset to. The object doesn't necessarily store what "unset" means for an object and it wouldn't have anything to roll back to.

**Comment by xilabao** [comment]
The progress of all the kubectl set subcommands is here. https://github.com/kubernetes/kubernetes/issues/21648#issuecomment-250881311. I think `unset` is useful for some resources.

**Comment by kargakis** [comment]
@kubernetes/sig-cli-feature-requests

**Comment by soltysh** [comment]
I'd see this being solve similarly to how allow clearing env vars, which is:
```
kubectl set env -f deploy.json ENV-
```
I don't think that a separate command is necessary. Eventually, where needed `--overwrite` should be added.
